src/main/java/edu/indiana/dlib/amppd/repository/PrimaryfileRepository.java

```diff
@@ -19,7 +19,7 @@
 	@Query(value = "select p from Primaryfile p where lower(p.name) like lower(concat('%', :keyword,'%')) or lower(p.description) like lower(concat('%', :keyword,'%'))")
 	List<Primaryfile> findByKeyword(@Param("keyword") String keyword); 
 		
-	@Query(value = "select p from Primaryfile p where ( lower(p.name) like lower(concat('%', :keyword,'%')) or lower(p.item.name) like lower(concat('%', :keyword,'%'))) order by p.item.id")
+	@Query(value = "select p from Primaryfile p where ( lower(p.name) like lower(concat('%', :keyword,'%')) or lower(p.item.name) like lower(concat('%', :keyword,'%')) or lower(p.item.collection.name) like lower(concat('%', :keyword,'%'))) order by p.item.id")
 	List<Primaryfile> findByItemOrFileName(@Param("keyword") String keyword);
 
```

Comment:
I suggest we leave the original findByItemOrFileName intact (might be useful in other context later), and add a new method findByCollectionOrItemOrFileName with your new query, and call this new method in your code.

Reply:
right, when you add the new method, you can change the references to the new method.

Reply:
tthe service name also looks Like findItemOrFile Do I need to change service name or create a new Service